core/src/main/java/org/apache/iceberg/util/PartitionUtil.java

```diff
@@ -51,6 +53,20 @@ private PartitionUtil() {
         MetadataColumns.FILE_PATH.fieldId(),
         convertConstant.apply(Types.StringType.get(), task.file().path()));
 
+    // add _spec
+    idToConstant.put(
+        MetadataColumns.SPEC.fieldId(),
+        convertConstant.apply(Types.IntegerType.get(), task.file().specId()));
+
+    // add _partition
+    if (partitionType.fields().isEmpty()) {
+      // use null as some query engines may not be able to handle empty structs
+      idToConstant.put(MetadataColumns.PARTITION_COLUMN_ID, null);
+    } else {
+      StructLike partition = Partitioning.reconcilePartition(partitionType, spec, partitionData);
+      idToConstant.put(MetadataColumns.PARTITION_COLUMN_ID, convertConstant.apply(partitionType, partition));
```

Comment:
Is `partitionType` derived from the read schema? It would be nice to be able to project.

Reply:
Actually, this probably doesn't matter because Spark will project just what it needs immediately and this is basically a constant for each file.

Reply:
You mean supporting schema pruning for metadata columns?

Reply:
I switched to using projections.